### PR TITLE
Scalable chat tags

### PR DESCRIPTION
**Pull request by Ved-s**
adds TextSnippet.UniqueDraw overload with Vector2 scale parameter
fixes ChatManager methods to use baseScale when rendering chat tags
fixes GetStringSize to work correctly with scaled chat tags

### Before

Scaling 0.7x
image

Scaling 2x
image

### After

Scaling 0.7x
image

Scaling 2x
image

### CODE REVIEW

File: patches/tModLoader/Terraria/UI/Chat/TextSnippet.cs.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
++using System;
+ 
+ namespace Terraria.UI.Chat
+ {
+@@ -36,6 +_,15 @@
+ 
+ 		public virtual Color GetVisibleColor() => ChatManager.WaveColor(Color);
+ 
++		/// By default, calls UniqueDraw with float scale parameter for compatibility
++		public virtual bool UniqueDraw(bool justCheckingString, out Vector2 size, SpriteBatch spriteBatch, Vector2 position = default(Vector2), Color color = default(Color), Vector2? scale = null) {
```
Comment (direwolf420) [inline patches/tModLoader/Terraria/UI/Chat/TextSnippet.cs.patch]: should be //

**Review by direwolf420 — COMMENTED** [review]
A before/after would be nice to see

File: patches/tModLoader/Terraria/UI/Chat/TextSnippet.cs.patch (hunk 1 above)
Comment (Ved-s) [reply patches/tModLoader/Terraria/UI/Chat/TextSnippet.cs.patch]: done

**Comment by JavidPack** [comment]
Looks like we arrived at some of the same conclusions. I was looking into this and found similar fixes, I'll do more testing later and see if we need to bring in some of the other changes I found when I was working on this earlier. 

Can you show how your current implementation works for the modconfig issue?
https://github.com/tModLoader/tModLoader/issues/2747

**Comment by Ved-s** [comment]
image
It's still off-center but it's not clipping out

**Comment by Ved-s** [comment]
Probably because there's a patting at the top because of wacky non-symmetrical font size

**Comment by JavidPack** [comment]
Thanks.

**Pull request merged**